### ISSUE TITLE
Exported Mono game: ERROR IN: res://.mono/assemblies/mscorlib.dll with _get_modified_time

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#19439]
**Godot version:**
3.0.3 RC3, Mono

**OS/device including version:**
Mageia 6 x86_64 (Linux), Mono 5.12.0.226 (self compiled).

**Issue description:**
Exporting a game works fine, and running it is functional too, but the terminal output is spammed with this error:
```
./test_mono.x86_64 
OpenGL ES 3.0 Renderer: GeForce GTX 670MX/PCIe/SSE2
Mono: Initializing module...
ERROR IN: res://.mono/assemblies/mscorlib.dll
ERROR: _get_modified_time: Method/Function Failed, returning: 0
   At: drivers/unix/file_access_unix.cpp:296.
ERROR IN: res://.mono/assemblies/mscorlib.dll
ERROR: _get_modified_time: Method/Function Failed, returning: 0
   At: drivers/unix/file_access_unix.cpp:296.
Mono: Runtime initialized
ERROR IN: res://.mono/assemblies/GodotSharp.dll
ERROR: _get_modified_time: Method/Function Failed, returning: 0
   At: drivers/unix/file_access_unix.cpp:296.
ERROR IN: res://.mono/assemblies/System.dll
ERROR: _get_modified_time: Method/Function Failed, returning: 0
   At: drivers/unix/file_access_unix.cpp:296.
ERROR IN: res://.mono/assemblies/v3_mono3.dll
ERROR: _get_modified_time: Method/Function Failed, returning: 0
   At: drivers/unix/file_access_unix.cpp:296.
Mono: INITIALIZED
Found class .Main for script res://Main.cs
Hello, I'm an exported Mono game!
ERROR IN: res://.mono/assemblies/System.Core.dll
ERROR: _get_modified_time: Method/Function Failed, returning: 0
   At: drivers/unix/file_access_unix.cpp:296.
```
The "Hello, I'm an exported Mono game!" is what my Main.cs's `_Ready` method prints, so that's functional.

CC @neikeq @hpvb 

**Steps to reproduce:**
- Download 3.0.3 RC3 Mono and install Mono templates
- Create a simple Mono project, build it
- Export (in my case to Linux X11 x86_64)
- Run the exported game in a terminal

**Minimal reproduction project:**
[v3_mono3.zip](https://github.com/godotengine/godot/files/2083153/v3_mono3.zip)

Test project used to trigger this issue. Includes the `.mono` and `.import` folders as used when exporting.

**Comment by akien-mga** [comment]
Exporting the data pack as ZIP to check the contents, here's what I get:
```
$ unzip ../test_mono.zip 
Archive:  ../test_mono.zip
  inflating: .mono/assemblies/GodotSharp.dll  
  inflating: .mono/assemblies/GodotSharpEditor.dll  
  inflating: .mono/assemblies/v3_mono3.dll  
replace .mono/assemblies/GodotSharp.dll? [y]es, [n]o, [A]ll, [N]one, [r]ename: y
  inflating: .mono/assemblies/GodotSharp.dll  
  inflating: .mono/assemblies/Mono.Posix.dll  
  inflating: .mono/assemblies/Mono.Security.dll  
  inflating: .mono/assemblies/System.dll  
  inflating: .mono/assemblies/System.Configuration.dll  
  inflating: .mono/assemblies/System.Core.dll  
  inflating: .mono/assemblies/System.Numerics.dll  
  inflating: .mono/assemblies/System.Security.dll  
  inflating: .mono/assemblies/System.Xml.dll  
  inflating: .mono/assemblies/mscorlib.dll  
  inflating: Main.cs                 
  inflating: MainNode.cs             
  inflating: Node.tscn               
  inflating: Node2D.tscn             
  inflating: Properties/AssemblyInfo.cs  
  inflating: default_env.tres        
  inflating: .import/icon.png-487276ed1e3a0c39cad0279d744ee560.stex  
  inflating: icon.png.import         
  inflating: icon.png                
  inflating: project.binary 
```

So:
- `GodotSharp.dll` is included twice in the archive?
- The DLLs that `_get_modified_time` complains about do exist, so the issue is not that they're missing.

On the other hand they seem to have a broken timestamp:
```
$ ls .mono/assemblies/ -l
total 12996
-rw-r--r-- 1 akien akien 1336832 Dec 31  1979 GodotSharp.dll
-rw-r--r-- 1 akien akien   38400 Dec 31  1979 GodotSharpEditor.dll
-rw-r--r-- 1 akien akien  212480 Dec 31  1979 Mono.Posix.dll
-rw-r--r-- 1 akien akien  309760 Dec 31  1979 Mono.Security.dll
-rw-r--r-- 1 akien akien  122368 Dec 31  1979 System.Configuration.dll
-rw-r--r-- 1 akien akien 1015296 Dec 31  1979 System.Core.dll
-rw-r--r-- 1 akien akien  123392 Dec 31  1979 System.Numerics.dll
-rw-r--r-- 1 akien akien  183808 Dec 31  1979 System.Security.dll
-rw-r--r-- 1 akien akien 3150336 Dec 31  1979 System.Xml.dll
-rw-r--r-- 1 akien akien 2797056 Dec 31  1979 System.dll
-rw-r--r-- 1 akien akien 3976192 Dec 31  1979 mscorlib.dll
-rw-r--r-- 1 akien akien    4096 Dec 31  1979 v3_mono3.dll
```

**Comment by akien-mga** [comment]
Similar errors with Windows 64-bit binary running in WINE:
```
$ wine test_mono.exe 
OpenGL ES 3.0 Renderer: GeForce GTX 670MX/PCIe/SSE2
fixme:xinput:XInputGetState (index 0, state 0x47fc514) Stub!
fixme:xinput:XInputGetStateEx (index 0, state 0x32bcd0) Stub!
fixme:win:RegisterTouchWindow (0x1004e 00000000): stub
Mono: Initializing module...
ERROR: find_mono: Cannot find mono in the registry
   At: modules/mono/utils/mono_reg_utils.cpp:161
no access to .mono/assemblies/mscorlib.dll
ERROR: _get_modified_time: Method/Function Failed, returning: 0
   At: drivers/windows/file_access_windows.cpp:290
no access to .mono/assemblies/mscorlib.dll
ERROR: _get_modified_time: Method/Function Failed, returning: 0
   At: drivers/windows/file_access_windows.cpp:290
Mono: Runtime initialized
no access to .mono/assemblies/GodotSharp.dll
ERROR: _get_modified_time: Method/Function Failed, returning: 0
   At: drivers/windows/file_access_windows.cpp:290
no access to .mono/assemblies/System.dll
ERROR: _get_modified_time: Method/Function Failed, returning: 0
   At: drivers/windows/file_access_windows.cpp:290
no access to .mono/assemblies/v3_mono3.dll
ERROR: _get_modified_time: Method/Function Failed, returning: 0
   At: drivers/windows/file_access_windows.cpp:290
Mono: INITIALIZED
Hello, I'm an exported Mono game!
no access to .mono/assemblies/System.Core.dll
ERROR: _get_modified_time: Method/Function Failed, returning: 0
   At: drivers/windows/file_access_windows.cpp:290
```

**Comment by neikeq** [comment]
I will fix this tonight. I guess the reason is that `get_modified_time` is not implemented for FileAccessPack. Not sure if it's a good thing to print an error because of that...

**Comment by neikeq** [comment]
I will also check why GodotSharp.dll is being exported twice.

**Comment by neikeq** [comment]
I can reproduce it on RC3. Where can I get the source to debug it?

**Comment by akien-mga** [comment]
@neikeq `git checkout 3.0`

**Comment by akien-mga** [comment]
3.0.3 RC3 might not be exactly the current `3.0/HEAD`, but it should be close enough. Otherwise @hpvb can maybe tell us which commit it was exactly. I'm currently building `3.0/HEAD` to see if I reproduce the bug with a self-compiled binary - otherwise the bug could come from the compilation itself.

**Comment by akien-mga** [comment]
I can reproduce it with a self-compiled `3.0/HEAD` on X11 too.

**Comment by neikeq** [comment]
There:

https://github.com/godotengine/godot/blob/e8bfc09b81f02395226a06f5a8ecd47bea136cc6/core/os/file_access.cpp#L480-L488

`FileAccess::get_modified_time` is missing the PackedData part at the beginning (see snippet in my other comment).

**Comment by neikeq** [comment]
@hpvb I labeled #16002 for cherry-picking.

**Comment by hpvb** [comment]
Fixed in 3.0

**Comment by neikeq** [comment]
@akien-mga #22688 should fix the issue you [commented](https://github.com/godotengine/godot/issues/19439#issuecomment-395663740) about where `GodotSharp.dll` is added twice to the archive.
